src/lib/components/NglMoleculeViewer.react.js

```diff
@@ -0,0 +1,274 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {Stage, Selection} from 'ngl';
+
+/**
+ * The NglMoleculeViewer is used to render schematic diagrams
+ * of biomolecules in ribbon-structure representations.
+ * Read more about the component here:
+ * https://github.com/IvoLeist/dash_ngl
+ * Read more about the used WebGL protein viewer here:
+ * https://github.com/arose/ngl
+ */
+export default class NglMoleculeViewer extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {stage: null, structuresList: []};
+    }
+
+    componentDidMount() {
+        const {id, stageParameters, viewportStyle} = this.props;
+        const params = {...stageParameters};
+        const stage = new Stage(id, params);
+        stage.setSize(viewportStyle.width, viewportStyle.height);
+        this.setState({stage});
+    }
+
+    shouldComponentUpdate(prevProps, nextProps) {
+        const {stageParameters, data} = this.props;
+
+        // check if data has changed
+        if ((data !== null) & (prevProps.data !== null)) {
+            // wait for the first pdb selection after startup
+            if (nextProps.data !== undefined) {
+                return true;
+            }
+
+            // check if pdb selection has changed
+            const oldSelection = prevProps.data[0].selectedValue;
+            const newSelection = data[0].selectedValue;
+            if (oldSelection !== newSelection) {
+                return true;
+            }
+        }
+
+        // check for stage params changed
+        const oldStage = prevProps.stageParameters;
+        const newStage = stageParameters;
+
+        // save it as a helper function in an extra script?
+        const isEqual = (obj1, obj2) => {
```

Comment:
You can use Ramda here, like I have done in the `fornaContainer` component: https://github.com/plotly/dash-bio/blob/c920884769f3a3b95d0836f3d5b4216258e0e5a6/src/lib/components/FornaContainer.react.js#L69

Reply:
Ah cool, never used it before. Works as expected :+1: